cmd/gazelle/fix-update.go

```diff
@@ -411,6 +436,37 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 	return exit
 }
 
+// lookupMapKindReplacement finds a mapped replacement for rule kind `kind`, resolving transitively.
+// i.e. if go_library is mapped to custom_go_library, and custom_go_library is mapped to other_go_library,
+// looking up go_library will return other_go_library.
+// It returns an error on a loop, and may return nil if no remapping should be performed.
+func lookupMapKindReplacement(kindMap map[string]config.MappedKind, kind string) (*config.MappedKind, error) {
```

Comment:
```suggestion
func lookupMapKindReplacement(kindMap map[string]config.MappedKind, kind string) (string, error) {
```
To simplify the interface, you can return the replacement kind here, because the caller of this function can get the result of info from `config.MappedKind`

Reply:
I'm not sure on this one - having a fully typed `*config.MappedKind` as a return value feels better documenting as to what the function is doing, and the caller is going to look up the output in `c.KindMap` again anyway, so it feels like returning a `*config.MappedKind` avoids potential errors such as a different KindMap being passed to the function and being used by the caller?